theurgy.lic

```diff
@@ -2,185 +2,453 @@
+    
+    @rituals.reject! do |r|
+      r[:items].any? { |i| !i[:on_hand] && !i[:shop] }
+    end
+    @communes.reject! do |c|
+      c[:items].any? { |i| !i[:on_hand] && !i[:shop] }
+    end
+  end
 
-  ######################## GET/PUT ########################
-
-  def put_in_container(item)
-    bput "put #{item} in my #{@theurgy_supply_container}", 'You put', 'What were you referring to'
+  def get_rituals
+    rituals = all_rituals
+    blacklist = @settings.theurgy_blacklist
+    rituals.reject! { |r| blacklist.include? r[:method].to_s }
+    whitelist = @settings.theurgy_whitelist
+    unless whitelist.empty?
+      rituals.select! { |r| whitelist.include? r[:method].to_s }
+    end
+    rituals
+  end
+  
+  def all_rituals
+    rituals = []
+    if @data['almsbox'] && @settings.tithe
+      rituals << {
+        method: :tithe,
+      }
+    end
+    if @hometown == 'Crossing'
+      if DRStats.circle >= 30
+        rituals << {
+          method: :study_wall,
+        }
+      end
+      rituals << {
+        method: :refectory,
+      }
+    end
+    if DRSkill.getrank('Mechanical Lore') > 140 && @immortal_aspect && @data['limb_foraging']
+      # TODO: allow other knife nouns 
+      rituals << {
+        method: :carve_bead,
+        items: [
+          { name: 'carving knife', any_container: true },
+          { name: 'shaper', any_container: true},
+          { name: 'prayer chain', any_container: true },
+        ],
+      }
+    end
+    if @data['gather_sirese'] && @data['plant_sirese']
+      rituals << {
+        method: :sirese_seed,
+      }
+    end
+    if @data['bath']
+      herbs = @data['herbs'] || ['sage', 'lavender']
+      rituals << {
+        method: :bathe,
+        items: herbs.map { |h| { name: h, shop: @data['theurgy_supplies'] } },
+      }
+    end
+    if @immortal_aspect
+      ritals << {
+        method: :meditate_bead,
+        items: [
+          { name: 'prayer chain', any_container: true },
+        ],
+      }
+    end
+    rituals << {
+      method: :pray_badge,
+    }
+    # TODO: prayer mat support
+    if @data['altar']
+      if DRStats.circle >= 10
+        rituals << {
+          method: :dance,
+        }
+      end
+      rituals << {
+        method: :incense,
+        items: [
+          { name: 'incense', shop: @data['theurgy_supplies'] },
+          { name: 'flint', shop: @data['flint_shop'] },
+        ]
+      }
+      rituals << {
+        method: :wine,
+        items: [
+          { name: 'wine', shop: @data['theurgy_supplies'] },
+        ],
+      }
+      rituals << {
+        method: :recite_prayer,
+      }
+      rituals << {
+        method: :clean_altar,
+      }
+    end
+    rituals << {
+      method: :embarass_myself,
+      items: [
+        {
+          name: 'golden parchment',
+          shop: @data['parchment_shop'],
+          purchase_method: :buy_parchment,
+        }
+      ],
+    }
+    
+    rituals
+  end
+  
+  def get_communes
```

Comment:
For all of the communes, I can't remember if you can't progress to or past said circle. Is it You can't be circle 2 unless you've completed the Tamsine quest, or you can't progress past circle 2 unless you have. I just flatly don't remember. So thought I'd leave that question here.

Reply:
I'm pretty sure it's "can't progress past."

Reply:
OK, thanks for the clarification.